### PR TITLE
DLMBL 2024 notebook

### PR DESCRIPTION
**Pull request by edyoshikun**
This PR adds the exercise for the DL@MBL 2024 course (Part 1) for the image translation. 
It also reorganizes the demo folders for clearer organization for the #119 wiki.

Major changes to our img2img translation example:

- Added the comparison to the pretrained VSCyto2D model
- Predictions and cellpose segmentations are computed and saved into OME-Zarr for further comparison (part 2 exercise)
- Compute pixel-based and segmentation-based metrics and stores them to a dataframe for further comparison.

### CODE REVIEW

**Comment by edyoshikun** [comment]
this PR should be merged before #119.

File: examples/virtual_staining/img2img_translation/setup.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+mkdir -p ~/data/06_image_translation/test
+
+# Change to the target directory
+cd ~/data/06_image_translation/training
+
+# Download the OME-Zarr dataset recursively
+wget -m -np -nH --cut-dirs=4 -R "index.html*" "https://public.czbiohub.org/comp.micro/viscy/VSCyto2D/training/a549_hoechst_cellmask_train_val.zarr/"
+cd ~/data/06_image_translation/test
+wget -m -np -nH --cut-dirs=4 -R "index.html*" "https://public.czbiohub.org/comp.micro/viscy/VSCyto2D/test/a549_hoechst_cellmask_test.zarr/"
+
```
Comment (edyoshikun) [inline examples/virtual_staining/img2img_translation/setup.sh:29]: we need to add the `VSCyto2D` model to public and here. @ziw-liu

**Comment by edyoshikun** [comment]
I am also updating cellpose to `3.0.10` in this PR. We were already using `3.0.1` so this shouldn't cause issues.

File: examples/virtual_staining/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+Examples scripts showcasing the usage of VisCy for different computer vision tasks.
+
+## Virtual staining
+
+### Image-to-Image translation using VisCy
+- [Image translation Exercise](./dlmbl_exercise/solution.py):
+Example showing how to use VisCy to train, predict and evaluate the VSCyto2D model. This notebook was developed for the [DL@MBL2024](https://github.com/dlmbl/DL-MBL-2024) course.
+
+- [Virtual staining exercise](./img2img_translation/solution.py): exploring the label-free to fluorescence virtual staining and florescence to label-free image translation task using VisCy UneXt2.
+
```
Comment (mattersoflight) [inline examples/virtual_staining/README.md:12]: As we discussed, these should be highlighted on the README at the root of repo.

File: examples/virtual_staining/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+### Running inference using models from the VisCy library
+The library of models can be found [here](https://github.com/mehta-lab/VisCy/wiki/Library-of-virtual-staining-(VS)-Models).
+
+- [Inference with VSCyto2D](./VS_model_inference/demo_vscyto2d.py):
+2D inference example on 20x A549 cell data. (Phase to nuclei and plasma membrane).
+- [Inference with VSCyto3D](./VS_model_inference/demo_vscyto3d.py):
+3D inference example on 63x HEK293T cell data. (Phase to nuclei and plasma membrane).
+- [Inference VSNeuromast](./VS_model_inference/demo_vsneuromast.py):
+3D inference example of 63x zebrafish neuromast data (Phase to nuclei and plasma membrane)
+
```
Comment (mattersoflight) [inline examples/virtual_staining/README.md:22]: As we discussed, detailed instructions to run inference and training with configs, scripts, and datasets should be documented on wiki; and linked on the README file at the root of repo.

**Review by mattersoflight — APPROVED** [review]
@edyoshikun I suggest that you improve the organization of files and documentation of demos (DL@MBL exercise, inference demos, instructions on how to run training with data/config/scripts we share) first. We can then iterate on the exercise.

**Comment by edyoshikun** [comment]
@ziw-liu merging this.

**Pull request merged**